core/services/offchainreporting/contract_tracker.go

```diff
@@ -0,0 +1,329 @@
+	q := ethereum.FilterQuery{
+		FromBlock: big.NewInt(int64(changedInBlock)),
+		ToBlock:   big.NewInt(int64(changedInBlock)),
+		Addresses: []gethCommon.Address{t.contractAddress},
+		Topics: [][]gethCommon.Hash{
+			{OCRContractConfigSet},
+		},
+	}
+
+	logs, err := t.ethClient.FilterLogs(ctx, q)
+	if err != nil {
+		return c, err
+	}
+	if len(logs) == 0 {
+		return c, errors.Errorf("ConfigFromLogs: OCRContract with address 0x%x has no logs", t.contractAddress)
+	}
+
+	latest, err := t.contractFilterer.ParseConfigSet(logs[len(logs)-1])
+	if err != nil {
+		return c, errors.Wrap(err, "ConfigFromLogs failed to ParseConfigSet")
+	}
+	latest.Raw = logs[len(logs)-1]
+	if latest.Raw.Address != t.contractAddress {
+		return c, errors.Errorf("log address of 0x%x does not match configured contract address of 0x%x", latest.Raw.Address, t.contractAddress)
+	}
+	return confighelper.ContractConfigFromConfigSetEvent(*latest), err
+}
+
+// LatestBlockHeight queries the eth node for the most recent header
+// TODO(sam): This could (should?) be optimised to use the head tracker
+// https://www.pivotaltracker.com/story/show/177006717
+func (t *OCRContractTracker) LatestBlockHeight(ctx context.Context) (blockheight uint64, err error) {
+	h, err := t.ethClient.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return 0, err
+	}
+	if h == nil {
+		return 0, errors.New("got nil head")
+	}
+
+	return uint64(h.Number), nil
+}
+
+// LatestRoundRequested returns the configDigest, epoch, and round from the latest
+// RoundRequested event emitted by the contract. LatestRoundRequested may or may not
+// return a result if the latest such event was emitted in a block b such that
+// b.timestamp < tip.timestamp - lookback.
+//
+// If no event is found, LatestRoundRequested should return zero values, not an error.
+// An error should only be returned if an actual error occurred during execution,
+// e.g. because there was an error querying the blockchain or the database.
+//
+// As an optimization, this function may also return zero values, if no
+// RoundRequested event has been emitted after the latest NewTransmission event.
+func (t *OCRContractTracker) LatestRoundRequested(ctx context.Context, lookback time.Duration) (configDigest ocrtypes.ConfigDigest, epoch uint32, round uint8, err error) {
+	// NOTE: This should be "good enough" 99% of the time.
+	// It guarantees validity up to `BLOCK_BACKFILL_DEPTH` blocks ago
+	// Some further improvements could be made:
+	// TODO: Can we increase the backfill depth?
+	// TODO: Can we use the lookback to optimise at all?
+	// TODO: How well can we satisfy the requirements after the latest round of changes to the log broadcaster?
+	// See: https://www.pivotaltracker.com/story/show/177063733
+	t.lrrMu.RLock()
+	defer t.lrrMu.RUnlock()
+	return t.latestRoundRequested.ConfigDigest, t.latestRoundRequested.Epoch, t.latestRoundRequested.Round, nil
```

Comment:
Could this ever be called when `t.latestRoundRequested` is an empty struct?

Reply:
Yes, and it returns zero values, which is in line with the requirement.